lib/widgets/gallery_view.dart

```diff
@@ -0,0 +1,258 @@
+import 'package:flutter/material.dart';
+import 'package:workoutpersonalizer_frontend/constants/styles.dart';
+
+class Gallery extends StatefulWidget {
+  final List<String> exerciseList;
+  final List<String> descriptionList;
+  final List<String> tagList;
+  final bool playlistGallery;
+
+  const Gallery(
+      {Key? key,
+      required this.exerciseList,
+      required this.playlistGallery,
+      required this.descriptionList,
+      required this.tagList})
+      : super(key: key);
+
+  @override
+  GalleryState createState() => GalleryState();
+}
+
+class GalleryState extends State<Gallery> {
+  List<String> exercisesToDisplay = [
+    'Pushup',
+    'Crunch',
+    'Neck Circles',
+    'Hamstring Stretch',
+    'Plank',
+    'Hip Flexor Stretch',
+  ];
+
+  void addToWorkouts(value) {
+    print("this is value " + value);
+  }
+
+  void addToLibrary() {
+    print("add to library");
+  }
+
+  List<String> getDisplayItems(text, exerciseList) {
+    List<String> ret = [];
+    exerciseList.forEach((String exercise) {
+      if (exercise.toLowerCase().contains(text)) {
+        ret.add(exercise);
+      }
+    });
+    return ret;
+  }
+
+  Widget buildGrid(BuildContext context, exerciseList, descriptionList, tagList,
+      playlistGallery) {
+    //TOO: figure out userEffect but in flutter to get rid of duplicate list
+    // exercisesToDisplay = exerciseList;
+    return GridView.builder(
+        gridDelegate: const SliverGridDelegateWithMaxCrossAxisExtent(
+            maxCrossAxisExtent: 300,
+            childAspectRatio: 8 / 7,
+            crossAxisSpacing: 20,
+            mainAxisSpacing: 20),
+        itemCount: exercisesToDisplay.length,
+        itemBuilder: (context, index) {
+          return GestureDetector(
+              onTap: () async {
+                await showDialog(
+                    context: context,
+                    builder: (_) => ExerciseDialog(
+                        exerciseList[index], descriptionList[index]));
+              },
+              child: Card(
+                semanticContainer: true,
+                clipBehavior: Clip.antiAliasWithSaveLayer,
+                child: Column(
```

Comment:
This is giving me an overflow error when it is resized to be too small. Try using `ListTile` here instead